### ISSUE TITLE
OneToOne relation with referencedColumnName does not work

### ISSUE THREAD

**Issue by JacobSoderblom** [typeorm/typeorm#341]
We are trying to create a OneToOne relation with referencedColumnName where the relation is not between primary keys, its between two unique strings. See example below

```
@Entity()
class Exercise {
  
  @PrimaryGeneratedColumn()
  public id: number;

  @Column('string')
  public key: string;

  @OneToOne((type) => Atom, (atom) => atom.exercise)
  public atom: Atom;
}

@Entity()
class Atom {

  @PrimaryGeneratedColumn()
  public id: number;

  @Column('string')
  public key: string;

  @Column('string')
  public exercise_key: string;

  @OneToOne((type) => Exercise, (exercise) => exercise.atom)
  @JoinColumn({ name: 'exercise_key',  referencedColumnName: 'key' })
  public exercise: Exercise
}
```
When typeorm tries to create the schema we get an error saying that MySQL could not create the foreign key. We debugged it and saw that `ForeignKeySchema.referencedColumnNames = ['id']` and not `ForeignKeySchema.referencedColumnNames = ['key']`, but I don't know if that's a part of the problem.

**Comment by JacobSoderblom** [comment]
@pleerock ping

**Comment by pleerock** [comment]
that was a bug. Fixed and released in `0.0.11`. Make sure to setup your entities exactly like in test in 08c1ee1e9e71f86ebeccd65e059ae5d2f204523a

**Comment by JacobSoderblom** [comment]
Thank you!


Mvh
Jacob Söderblom
________________________________
Från: Umed Khudoiberdiev <notifications@github.com>
Skickat: den 24 mars 2017 13:26:19
Till: typeorm/typeorm
Kopia: Jacob Söderblom; Author
Ämne: Re: [typeorm/typeorm] OneToOne relation with referencedColumnName does not work (#341)


that was a bug. Fixed and released in 0.0.11. Make sure to setup your entities exactly like in test in 08c1ee1<https://github.com/typeorm/typeorm/commit/08c1ee1e9e71f86ebeccd65e059ae5d2f204523a>
